server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -1331,6 +1349,11 @@ public ClusterState getClusterStateForManifest(
                 false,
                 includeEphemeral
             );
+
+            if (checksumValidationEnabled && manifest.getClusterStateChecksum()!=null) {
```

Comment:
Should we have an separate check for `CODEC_VERSION`? Also would be good to place it in the end rather than in a specific if condition to avoid any cases being missed in future.

Reply:
Didnt get this, what do you mean in the end? Currently we are adding this validation when we create the cluster state after reading from manifest.
Codec check is not required since we are checking for checksum being null in manifest.